earth_enterprise/src/fusion/autoingest/tools/geaddtoimageryproject.cpp

```diff
@@ -151,21 +173,39 @@ main(int argc, char *argv[]) {
         (mercator? kMercatorProjectSubtype : kProjectSubtype));
 
     // TimeMachine only applies to non-mercator
-     if (!mercator) {
+    if (!mercator) {
       req.enable_timemachine = enable_historical_imagery;
       req.disable_timemachine = disable_historical_imagery;
     }
 
+    // Only pass along the overlay-related options if it's a terrain project
+    if (AssetDefs::Terrain == AssetType){
+      req.enable_terrain_overlay = enable_terrain_overlay;
+      req.disable_terrain_overlay = disable_terrain_overlay;
+      req.overlay_terrain_start_level = start_level;
+      req.overlay_terrain_resources_min_level = resource_min_level;
+    }
+
     if (req.items.empty()) {
       if (cmd == &khAssetManagerProxy::RasterProjectNew ||
           cmd == &khAssetManagerProxy::MercatorRasterProjectNew) {
         notify(NFY_WARN,
                "No insets specified. Project will be empty.");
       } else if (mercator ||
-            !(enable_historical_imagery || disable_historical_imagery)) {
-        // For adding/modifying, we always need insets except:
-        // non-mercator when historical_imagery/no_historical_imagery is
-        // specified.
+        !( (AssetDefs::Imagery == AssetType && (enable_historical_imagery || disable_historical_imagery)) ||
+           (AssetDefs::Terrain == AssetType && 
+              (enable_terrain_overlay || disable_terrain_overlay || 
+              start_level != 0 || resource_min_level != 0))
+        )){
+            // !(AssetDefs::Imagery == AssetType && (enable_historical_imagery || disable_historical_imagery)) ||
```

Comment:
Executing `genewimageryproject --help` output information about terrain overlay. I think even if imagery and terrain are using common code we probably should not output specific terrain info when dealing with imagery.
Here is the specific terrain overlay info i get:
--> ```--terrain_overlay : make this terrain project an overlay project
      --no_terrain_overlay : make this terrain project a normal project
      --start_level : the level from which to start building the terrain overlay project
      --resource_min_level : the threshold level that separates fill terrain from overlay terrain
   By default, new terrain projects are NOT overlay projects unless --terrain_overlay is specified.```

I get the same behavior when i run `gemodifyimageryproject --help`

Reply:
I wrestled with this a bit while writing the code. On the one hand, the original developers set a precedent of including the same usage text for all commands that are handled by geaddtoimageryproject.cpp. That means that even `ge***terrainproject` would list `--historical_imagery` and `--no_historical_imagery` in their `--help` text.

I thought about adding logic to only display appropriate usage text but decided to leave it as is for now. However, that's probably not hard to change. I'll take a stab at separating the imagery and terrain project usage text.